pkg/domain/infra/abi/farm.go

```diff
@@ -0,0 +1,128 @@
+package abi
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+
+	"github.com/containers/buildah/pkg/parse"
+	"github.com/containers/common/libimage"
+	istorage "github.com/containers/image/v5/storage"
+	"github.com/containers/podman/v4/pkg/domain/entities"
+	"github.com/containers/podman/v4/pkg/emulation"
+)
+
+// FarmName returns the local engine's name.
+func (ir *ImageEngine) FarmName(ctx context.Context) string {
+	return entities.LocalFarmImageBuilderName
+}
+
+// FarmDriver returns a description of the local image builder driver
+func (ir *ImageEngine) FarmDriver(ctx context.Context) string {
+	return entities.LocalFarmImageBuilderDriver
+}
+
+func (ir *ImageEngine) fetchInfo(_ context.Context) (os, arch, variant string, nativePlatforms []string, emulatedPlatforms []string, err error) {
+	nativePlatform := parse.DefaultPlatform()
+	platform := strings.SplitN(nativePlatform, "/", 3)
+	switch len(platform) {
+	case 0, 1:
+		return "", "", "", nil, nil, fmt.Errorf("unparseable default platform %q", nativePlatform)
+	case 2:
+		os, arch = platform[0], platform[1]
+	case 3:
+		os, arch, variant = platform[0], platform[1], platform[2]
+	}
+	os, arch, variant = libimage.NormalizePlatform(os, arch, variant)
+	nativePlatform = os + "/" + arch
+	if variant != "" {
+		nativePlatform += ("/" + variant)
+	}
+	emulatedPlatforms = emulation.Registered()
+	return os, arch, variant, append([]string{}, nativePlatform), emulatedPlatforms, nil
+}
+
+// FarmInspect returns information about the remote engines in the farm
+func (ir *ImageEngine) FarmInspect(ctx context.Context) (*entities.FarmInspectReport, error) {
+	var (
+		platforms          sync.Once
+		platformsErr       error
+		os, arch, variant  string
+		nativeP, emulatedP []string
+	)
+	platforms.Do(func() {
```

Comment:
It looks like `platforms.Do` is always going to call the passed-in function.

Reply:
fixed